openvdb/points/PointDataGrid.h

```diff
@@ -999,13 +999,20 @@ template<typename ValueIterT, typename FilterT>
 inline IndexIter<ValueIterT, FilterT>
 PointDataLeafNode<T, Log2Dim>::beginIndex(const FilterT& filter) const
 {
+    FilterT newFilter(filter);
+    newFilter.reset(*this);
+
+    // generate no-op iterator if filter evaluates no indices
+
+    if (filter.state() == index::NONE) {
```

Comment:
Can move this up before the filter copy, though this still dosen't solve the double copy problem